libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -41,46 +43,94 @@ internal class RouteAlternativesController constructor(
+            ) {
+                routeAlternativesObserver.onRouteAlternatives(
+                    routeProgress,
+                    alternatives.toDirectionsRoutes(),
+                    routerOrigin
+                )
+            }
+
+            override fun onRouteAlternativesError(error: RouteAlternativesError) {
+                logE(TAG, Message("Error: ${error.message}"))
+            }
+        }
+        legacyObserversMap[routeAlternativesObserver] = observer
+        register(observer)
+    }
+
+    fun unregister(routeAlternativesObserver: RouteAlternativesObserver) {
+        val observer = legacyObserversMap.remove(routeAlternativesObserver)
+        if (observer != null) {
+            unregister(observer)
+        }
+    }
+
+    fun register(routeAlternativesObserver: NavigationRouteAlternativesObserver) {
         val isStopped = observers.isEmpty()
         observers.add(routeAlternativesObserver)
         if (isStopped) {
             nativeRouteAlternativesController.addObserver(nativeObserver)
         }
     }
 
-    fun unregister(routeAlternativesObserver: RouteAlternativesObserver) {
+    fun unregister(routeAlternativesObserver: NavigationRouteAlternativesObserver) {
         observers.remove(routeAlternativesObserver)
         if (observers.isEmpty()) {
             nativeRouteAlternativesController.removeObserver(nativeObserver)
         }
     }
 
-    fun triggerAlternativeRequest(listener: RouteAlternativesRequestCallback?) {
+    fun triggerAlternativeRequest(listener: NavigationRouteAlternativesRequestCallback?) {
         nativeRouteAlternativesController.refreshImmediately { expected ->
             val routeProgress = tripSession.getRouteProgress()
                 ?: run {
-                    listener?.onRouteAlternativesAborted(
-                        """
-                            |Route progress not available, ignoring alternatives update.
-                            |Continuous alternatives are only available in active guidance.
-                        """.trimMargin()
+                    listener?.onRouteAlternativesRequestError(
+                        RouteAlternativesError(
+                            message =
+                            """
+                                |Route progress not available, ignoring alternatives update.
+                                |Continuous alternatives are only available in active guidance.
+                            """.trimMargin()
+                        )
                     )
                     return@refreshImmediately
                 }
 
             expected.fold(
                 { error ->
-                    listener?.onRouteAlternativesAborted(error)
+                    listener?.onRouteAlternativesRequestError(
+                        // NN should expose origin of a failed alternatives request,
+                        // refs https://github.com/mapbox/mapbox-navigation-native/issues/5401
```

Comment:
Capturing from @DenisPryt 
> RouteAlternative.route.routerOrigin: optionalrouter::RouterOrigin
Empty optional means that route from setRoutes method (unknown origin)

Reply:
This is not yet available in Nav Native, it was just merged and I assume we'll be able to address this with the upcoming release.